src/components/pages/landing/Gooddollar.js

```diff
@@ -29,102 +29,99 @@ export const Gooddollar = ({ setShowGooddollarVerification }) => {
-            'users',
-            updateData,
-            { wallet_identifier: wallet.accountId }
-          );
-        } else {
-          const { error: appError } = await supabase.insert(
-            'users',
-            updateData
+        try {
+          const result = await verifyUser(sendObj);
+          const { data } = await supabase.select('users', {
+            wallet_identifier: wallet.accountId,
+          });
+
+          if (data?.[0]) {
+            const { error: appError } = await supabase.update(
+              'users',
+              updateData,
+              { wallet_identifier: wallet.accountId }
+            );
+          } else {
+            const { error: appError } = await supabase.insert(
+              'users',
+              updateData
+            );
+          }
+          console.log(result);
+          if (result?.m) {
+            await wallet.callMethod({
+              contractId: gooddollar_contract,
+              method: 'sbt_mint',
+              args: {
+                claim_b64: result.m,
+                claim_sig: result.sig,
+              },
+              deposit: '8000000000000000000000',
+            });
+            log_event({ event_log: 'Applied for OG SBT' });
+          } else {
+            setIsVerifyApiError(true);
+          }
+        } catch (e) {
+          console.log('Error', e);
+          toast.error(
+            'An error occured while submitting your details , please try again'
           );
+        } finally {
+          setSubmitting(false);
         }
-        console.log(result);
-        await wallet.callMethod({
-          contractId: gooddollar_contract,
-          method: 'sbt_mint',
-          args: {
-            claim_b64: result.m,
-            claim_sig: result.sig,
-          },
-          deposit: '8000000000000000000000',
-        });
-        log_event({ event_log: 'Applied for OG SBT' });
-      } catch (e) {
-        console.log('Error', e);
-        toast.error(
-          'An error occured while submitting your details , please try again'
-        );
-      } finally {
-        setSubmitting(false);
-      }
-    },
-    validate: (values) => {
-      const errors = {};
-      // if (values.phone) {
-      //   if (!isPossiblePhoneNumber(values.phone)) {
-      //     errors.phone =
-      //       "Invalid phone number, please provide the phone number with valid country code";
-      //   }
-      // } else {
-      //   errors.phone = "Phone Number Required";
-      // }
-      return errors;
-    },
-  });
+      },
+      validate: (values) => {
+        const errors = {};
+        // if (values.phone) {
+        //   if (!isPossiblePhoneNumber(values.phone)) {
+        //     errors.phone =
+        //       "Invalid phone number, please provide the phone number with valid country code";
+        //   }
+        // } else {
+        //   errors.phone = "Phone Number Required";
+        // }
```

Comment:
can you add the reason about why we have commented it

Reply:
We are not using the user phone number any more hence validation not needed